File: android/src/main/java/com/risemaxi/graft/classes/results/GetInstallIdResult.java
package com.risemaxi.graft.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import com.risemaxi.graft.interfaces.Result;

public class GetInstallIdResult implements Result {

    private final int bucket;

    @NonNull
    private final String installId;

    public GetInstallIdResult(@NonNull String installId, int bucket) {
        this.installId = installId;
        this.bucket = bucket;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("bucket", bucket);
        result.put("installId", installId);
        return result;
    }
}

File: ios/Sources/RisemaxiGraft/Classes/Results/GetInstallIdResult.swift
import Foundation
import Capacitor

@objc public class GetInstallIdResult: NSObject, Result {
    let bucket: Int
    let installId: String

    init(installId: String, bucket: Int) {
        self.installId = installId
        self.bucket = bucket
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["bucket"] = bucket
        result["installId"] = installId
        return result as AnyObject
    }
}
